support/sier-codec/src/lib.rs

```diff
@@ -48,13 +50,79 @@ impl Parser {
     pub fn struct_def(&self, name: &str) -> Option<&Arc<StructDef>> {
         self.structs.values().find(|s| s.type_name() == name)
     }
-}
 
-impl Default for Parser {
-    fn default() -> Self {
-        Parser {
-            structs: HashMap::new(),
+    fn transform_serde_value<'a>(
+        &mut self,
+        value: &SerdeValue,
+    ) -> Result<(Type, Value<'a>), Error<'a>> {
+        match value {
+            SerdeValue::Null => Ok((Type::Unit, Value::Unit)),
+            SerdeValue::Bool(_bool) => Ok((Type::Unit, Value::Unit)),
+            SerdeValue::Number(number) => {
+                if let Some(n) = number.as_u64() {
+                    Ok((Type::U64, Value::U64(n)))
+                } else {
+                    Ok((Type::Unit, Value::Unit))
```

Comment:
We should error or return float? I think it's very confusing to return Unit when an error happens. While developing, you can leave `todo!()` or `unimplemented!("<details>")` for cases you haven't covered yet.